dask/array/utils.py

```diff
@@ -69,18 +75,24 @@ def assert_eq(a, b, check_shape=True, **kwargs):
         a = a.compute(get=get_sync)
         if hasattr(a, 'todense'):
             a = a.todense()
+        if not hasattr(a, 'dtype'):
+            a = np.array(a)
         if _not_empty(a):
             assert a.dtype == a_original.dtype
         if check_shape:
             assert_eq_shape(a_original.shape, a.shape, check_nan=False)
     else:
         adt = getattr(a, 'dtype', None)
+        if not hasattr(a, 'dtype'):
+            a = np.array(a)
```

Comment:
The only time that things should pass through without a dtype attribute (which is on all numpy scalars) is if it's an object dtype. At that, I'd switch the logic to:

```
if not hasattr(a, 'dtype'):
    a = np.array(a, dtype='O')
```

You'll also want to move this above the `adt = getattr(a, 'dtype', None)`, as we'll probably want `assert_eq(np.array(1, dtype='O'), 1)` to pass.